components/Tabs/tests/unit/Theming/TabsThemingTests.m

```diff
@@ -0,0 +1,197 @@
+  self.colorScheme.onSurfaceColor = [UIColor colorWithWhite:(CGFloat)0.1 alpha:(CGFloat)0.8];
+  self.colorScheme.onBackgroundColor = [UIColor colorWithWhite:0 alpha:(CGFloat)0.9];
+
+  self.typographyScheme.headline1 = [UIFont systemFontOfSize:1];
+  self.typographyScheme.headline2 = [UIFont systemFontOfSize:2];
+  self.typographyScheme.headline3 = [UIFont systemFontOfSize:3];
+  self.typographyScheme.headline4 = [UIFont systemFontOfSize:4];
+  self.typographyScheme.headline5 = [UIFont systemFontOfSize:5];
+  self.typographyScheme.headline6 = [UIFont systemFontOfSize:6];
+  self.typographyScheme.subtitle1 = [UIFont systemFontOfSize:7];
+  self.typographyScheme.subtitle2 = [UIFont systemFontOfSize:8];
+  self.typographyScheme.body1 = [UIFont systemFontOfSize:9];
+  self.typographyScheme.body2 = [UIFont systemFontOfSize:10];
+  self.typographyScheme.caption = [UIFont systemFontOfSize:11];
+  self.typographyScheme.button = [UIFont systemFontOfSize:12];
+  self.typographyScheme.overline = [UIFont systemFontOfSize:13];
+
+  self.containerScheme.colorScheme = self.colorScheme;
+  self.containerScheme.typographyScheme = self.typographyScheme;
+  [self.tabBar applyPrimaryThemeWithScheme:self.containerScheme];
+
+  // Then
+  [self verifyTabBarPrimaryTheming];
+}
+
+- (void)verifyTabBarPrimaryTheming {
+  // Color
+  XCTAssertEqualObjects(self.tabBar.barTintColor, self.colorScheme.primaryColor);
+  XCTAssertEqualObjects(self.tabBar.tintColor, self.colorScheme.onPrimaryColor);
+  XCTAssertEqualObjects([self.tabBar titleColorForState:MDCTabBarItemStateSelected],
+                        self.colorScheme.onPrimaryColor);
+  XCTAssertEqualObjects([self.tabBar imageTintColorForState:MDCTabBarItemStateSelected],
+                        self.colorScheme.onPrimaryColor);
+  UIColor *unselectedTitleColor =
+      [self.colorScheme.onPrimaryColor colorWithAlphaComponent:kUnselectedTitleOpacity];
+  UIColor *unselectedImageColor =
+      [self.colorScheme.onPrimaryColor colorWithAlphaComponent:kUnselectedImageOpacity];
+  XCTAssertEqualObjects([self.tabBar titleColorForState:MDCTabBarItemStateNormal],
+                        unselectedTitleColor);
+  XCTAssertEqualObjects([self.tabBar imageTintColorForState:MDCTabBarItemStateNormal],
+                        unselectedImageColor);
+  XCTAssertEqualObjects(
+      self.tabBar.bottomDividerColor,
+      [self.colorScheme.onPrimaryColor colorWithAlphaComponent:kBottomDividerOpacity]);
+
+  // Typography
+  XCTAssertEqualObjects(self.tabBar.selectedItemTitleFont, self.typographyScheme.button);
+  XCTAssertEqualObjects(self.tabBar.unselectedItemTitleFont, self.typographyScheme.button);
+}
+
+- (void)testTabBarSurfaceVariantThemingDefault {
+  // When
+  [self.tabBar applySurfaceVariantThemeWithScheme:self.containerScheme];
+
+  // Then
+  [self verifyTabBarSurfaceVariantTheming];
+}
+
+- (void)testTabBarSurfaceVariantThemingCustom {
```

Comment:
Tip: Please move the helpers near the top of the class so they are together with other non-test methods. Then you can add a `#pragma` to mark the start of the actual tests for easier reading later.